openedx/core/djangoapps/waffle_utils/__init__.py

```diff
@@ -1,503 +1,65 @@
 """
-Utilities for waffle.
-
-Includes namespacing, caching, and course overrides for waffle flags.
-
-Usage:
-
-For Waffle Flags, first set up the namespace, and then create flags using the
-namespace.  For example::
-
-   WAFFLE_FLAG_NAMESPACE = WaffleFlagNamespace(name='my_namespace')
-
-   # Use CourseWaffleFlag when you are in the context of a course.
-   SOME_COURSE_FLAG = CourseWaffleFlag(WAFFLE_FLAG_NAMESPACE, 'some_course_feature', __name__)
-   # Use WaffleFlag when outside the context of a course.
-   SOME_FLAG = WaffleFlag(WAFFLE_FLAG_NAMESPACE, 'some_feature', __name__)
-
-You can check these flags in code using the following::
-
-    SOME_FLAG.is_enabled()
-    SOME_COURSE_FLAG.is_enabled(course_key)
-
-To test these WaffleFlags, see testutils.py.
-
-In the above examples, you will use Django Admin "waffle" section to configure
-for a flag named: my_namespace.some_course_feature
-
-You could also use the Django Admin "waffle_utils" section to configure a course
-override for this same flag (e.g. my_namespace.some_course_feature).
-
-For Waffle Switches, first set up the namespace, and then create the flag name.
-For example::
-
-    WAFFLE_SWITCHES = WaffleSwitchNamespace(name=WAFFLE_NAMESPACE)
-
-    ESTIMATE_FIRST_ATTEMPTED = 'estimate_first_attempted'
-
-You can then use the switch as follows::
-
-    WAFFLE_SWITCHES.is_enabled(waffle.ESTIMATE_FIRST_ATTEMPTED)
-
-To test WaffleSwitchNamespace, use the provided context managers.  For example:
-
-    with WAFFLE_SWITCHES.override(waffle.ESTIMATE_FIRST_ATTEMPTED, active=True):
-        ...
-
-For long-lived flags, you may want to change the default for devstack, sandboxes,
-or new Open edX releases. For help with this, see:
-openedx/core/djangoapps/waffle_utils/docs/decisions/0001-refactor-waffle-flag-default.rst
-
-Also see ``WAFFLE_FLAG_CUSTOM_ATTRIBUTES`` and docstring for _set_waffle_flag_attribute
-for temporarily instrumenting/monitoring waffle flag usage.
-
+Extra utilities for waffle: most classes are defined in edx_toggles.toggles (https://edx-toggles.readthedocs.io/), but
+we keep here some extra classes for usage within edx-platform. These classes cover course override use cases.
 """
-
 import logging
-import re
-import traceback
-from abc import ABCMeta
-from contextlib import contextmanager
-from weakref import WeakSet
 
-import crum
-import six
-from django.conf import settings
-from edx_django_utils.monitoring import set_custom_attribute
 from opaque_keys.edx.keys import CourseKey
-from waffle import flag_is_active, switch_is_active
 
-from openedx.core.lib.cache_utils import get_cache as get_request_cache
+from edx_toggles.toggles import WaffleFlag, WaffleFlagNamespace, WaffleSwitch, WaffleSwitchNamespace
```

Comment:
Reminder: Please add the wrappers with `set_custom_attribute`.  Thanks.

Reply:
:ok_hand: